static/js/modules/tables.js

```diff
@@ -396,127 +349,197 @@ function adjustFormHeight($table, $form) {
+};
+
+var defaultCallbacks = {
+  afterRender: function() {}
+};
+
+function DataTable(selector, opts) {
+  opts = opts || {};
+  this.$body = $(selector);
+  this.opts = _.extend({}, defaultOpts, {ajax: this.fetch.bind(this)}, opts);
+  this.callbacks = _.extend({}, defaultCallbacks, opts.callbacks);
+  this.filterSet = (this.opts.panel || {}).filterSet;
+
+  this.xhr = null;
+  this.fetchContext = null;
+  this.hasWidgets = null;
+  this.filters = null;
+
+  var Paginator = this.opts.paginator || OffsetPaginator;
+  this.paginator = new Paginator();
+  this.api = this.$body.DataTable(this.opts);
+
+  DataTable.registry[this.$body.attr('id')] = this;
+
+  if (this.filterSet) {
+    updateOnChange(this.filterSet.$body, this.api);
+    updateQuery(this.filterSet.serialize(), this.filterSet.fields);
+    this.$body.on('draw.dt', adjustFormHeight.bind(null, this.$body, this.filterSet.$body));
   }
-  opts = _.extend({
-    serverSide: true,
-    searching: false,
-    columns: columns,
-    lengthMenu: [30, 50, 100],
-    responsive: {
-      details: false
-    },
-    language: {
-      lengthMenu: 'Results per page: _MENU_'
-    },
-    dom: '<"results-info results-info--top"lfrp><"panel__main"t><"results-info"ip>',
-    ajax: function(data, callback, settings) {
-      var api = this.api();
-      if ($form) {
-        pushQuery(filterPanel.filterSet.serialize(), filterPanel.filterSet.fields);
-        parsedFilters = filterPanel.filterSet.serialize();
-      }
-      var query = _.extend(
-        callbacks.mapQuery(api, data),
-        parsedFilters || {}
-      );
-      if (useHideNull) {
-        query = _.extend(
-          query,
-          {sort_hide_null: $hideNullWidget.is(':checked')}
-        );
-      }
-      query.sort = mapSort(data.order, columns);
-      $processing.show();
-      $.getJSON(
-        helpers.buildUrl(path, _.extend({}, query, baseQuery || {}))
-      ).done(function(response) {
-        callbacks.handleResponse(api, data, response);
-        callback(callbacks.preprocess(response));
-        callbacks.afterRender(api, data, response);
-        if (opts.hideEmpty) {
-          hideEmpty(api, data, response);
-        }
-      }).always(function() {
-        $processing.hide();
-      });
-    }
-  }, opts || {});
-  var api = $table.DataTable(opts);
-  callbacks = _.extend({
-    handleResponse: function() {},
-    afterRender: function() {}
-  }, callbacks);
-  if (useFilters) {
-    // Update filters and data table on navigation
-    $(window).on('popstate', function() {
-      filterPanel.filterSet.activate();
-      var tempFilters = filterPanel.filterSet.serialize();
-      if (!_.isEqual(tempFilters, parsedFilters)) {
-        api.ajax.reload();
-      }
-    });
+
+  if (this.opts.useFilters) {
+    $(window).on('popstate', this.handlePopState.bind(this));
+  }
+
+  this.$body.css('width', '100%');
+  this.$body.find('tbody').addClass('js-panel-toggle');
```

Comment:
The `tbody` isn't in the HTML--it's drawn by DataTables on AJAX complete. I'm not sure about the `width` call. @noahmanger, do you remember where that came from? This refactor basically takes our existing code and bundles it up into classes, including some bits, like this, that might benefit from more revision.